pkgs/by-name/te/telegraph/package.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, desktop-file-utils
+, fetchFromGitHub
+, gobject-introspection
+, gtk4
+, libadwaita
+, meson
+, ninja
+, pkg-config
+, python3
+, stdenv
+, wrapGAppsHook4
+}:
+
+stdenv.mkDerivation rec {
+  pname = "telegraph";
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "fkinoshita";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-m36YHIo1PaDunnC12feSAbwwG1+E7s90fzOKskHtIag=";
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    gobject-introspection
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    gtk4
+    libadwaita
+    (python3.withPackages (ps: with ps; [
+      pygobject3
+    ]))
+  ];
+
+  meta = with lib; {
+    description = "Write and decode Morse";
+    homepage = "https://github.com/fkinoshita/Telegraph";
+    changelog = "https://github.com/fkinoshita/Telegraph/releases/v${version}";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ michaelgrahamevans ];
+    mainProgram = "telegraph";
+    platforms = platforms.linux;
+  };
+}
```

Comment:
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "telegraph";
  version = "0.1.8";

  src = fetchFromGitHub {
    owner = "fkinoshita";
    repo = "telegraph";
    rev = "v${finalAttrs.version}";
    hash = "sha256-m36YHIo1PaDunnC12feSAbwwG1+E7s90fzOKskHtIag=";
  };

  nativeBuildInputs = [
    desktop-file-utils
    gobject-introspection
    meson
    ninja
    pkg-config
    wrapGAppsHook4
  ];

  buildInputs = [
    gtk4
    libadwaita
    (python3.withPackages (ps: with ps; [
      pygobject3
    ]))
  ];

  meta = with lib; {
    description = "Write and decode Morse";
    homepage = "https://github.com/fkinoshita/Telegraph";
    changelog = "https://github.com/fkinoshita/Telegraph/releases/v${finalAttrs.version}";
    license = licenses.gpl3Only;
    maintainers = with maintainers; [ michaelgrahamevans ];
    mainProgram = "telegraph";
    platforms = platforms.linux;
  };
})
```

Reply:
I did not know about the finalAttrs pattern until now, thanks! :)